### PR TITLE
Fix docstring for `SeawaterBuoyancy`

### PR DESCRIPTION
**Pull request by glwagner**
Noticed while working on #3329

### CODE REVIEW

File: src/BuoyancyModels/seawater_buoyancy.jl [hunk 1]
```diff
@@ -43,22 +43,39 @@ end
     SeawaterBuoyancy([FT = Float64;]
                      gravitational_acceleration = g_Earth,
                      equation_of_state = LinearEquationOfState(FT),
-                     constant_temperature = false,
-                     constant_salinity = false)
+                     constant_temperature = nothing,
+                     constant_salinity = nothing)
 
 Returns parameters for a temperature- and salt-stratified seawater buoyancy model
```
Comment (navidcy) [inline src/BuoyancyModels/seawater_buoyancy.jl:49]:
```suggestion
Return parameters for a temperature- and salt-stratified seawater buoyancy model
```

File: src/BuoyancyModels/seawater_buoyancy.jl [hunk 2]
```diff
@@ -43,22 +43,39 @@ end
...
+teos10 = TEOS10EquationOfState()
+
+# Buoyancy that depends on both temperature and salinity
+buoyancy = SeawaterBuoyancy(equation_of_state=teos10)
+
+# Buoyancy that depends only on salinity with temperature held at 20 degrees Celsius
+salinity_dependent_buoyancy = SeawaterBuoyancy(equation_of_state=teos10, constant_temperature=20) 
+
+# Buoyancy that depends only on temperature with salinity held at 35 psu
+temperature_dependent_buoyancy = SeawaterBuoyancy(equation_of_state=teos10, constant_salinity=35)
```
Comment (navidcy) [inline src/BuoyancyModels/seawater_buoyancy.jl:77]:
```suggestion
julia> using Oceananigans
julia> using SeawaterPolynomials.TEOS10: TEOS10EquationOfState

# The "TEOS10" equation of state, see https://www.teos-10.org
julia> teos10 = TEOS10EquationOfState()
BoussinesqEquationOfState{Float64}:
    ├── seawater_polynomial: TEOS10SeawaterPolynomial{Float64}
    └── reference_density: 1020.0

# Buoyancy that depends on both temperature and salinity
julia> buoyancy = SeawaterBuoyancy(equation_of_state=teos10)
SeawaterBuoyancy{Float64}:
├── gravitational_acceleration: 9.80665
└── equation of state: BoussinesqEquationOfState{Float64}

# Buoyancy that depends only on salinity with temperature held at 20 degrees Celsius
julia> salinity_dependent_buoyancy = SeawaterBuoyancy(equation_of_state=teos10, constant_temperature=20)
SeawaterBuoyancy{Float64}:
├── gravitational_acceleration: 9.80665
├── constant_temperature: 20
└── equation of state: BoussinesqEquationOfState{Float64}

# Buoyancy that depends only on temperature with salinity held at 35 psu
julia> temperature_dependent_buoyancy = SeawaterBuoyancy(equation_of_state=teos10, constant_salinity=35)
SeawaterBuoyancy{Float64}:
├── gravitational_acceleration: 9.80665
├── constant_salinity: 35
└── equation of state: BoussinesqEquationOfState{Float64}
```

**Review by navidcy — APPROVED** [review]
I suggested to convert the example to doctest

File: src/BuoyancyModels/seawater_buoyancy.jl (hunk 2 above)
Comment (navidcy) [inline src/BuoyancyModels/seawater_buoyancy.jl:77]:
```suggestion
```jldoctest
Suggested change
using SeawaterPolynomials.TEOS10: TEOS10EquationOfState
# The "TEOS10" equation of state, see https://www.teos-10.org
teos10 = TEOS10EquationOfState()
# Buoyancy that depends on both temperature and salinity
buoyancy = SeawaterBuoyancy(equation_of_state=teos10)
# Buoyancy that depends only on salinity with temperature held at 20 degrees Celsius
salinity_dependent_buoyancy = SeawaterBuoyancy(equation_of_state=teos10, constant_temperature=20) 
# Buoyancy that depends only on temperature with salinity held at 35 psu
temperature_dependent_buoyancy = SeawaterBuoyancy(equation_of_state=teos10, constant_salinity=35)
julia> using Oceananigans
julia> using SeawaterPolynomials.TEOS10: TEOS10EquationOfState
# The "TEOS10" equation of state, see https://www.teos-10.org
julia> teos10 = TEOS10EquationOfState()
BoussinesqEquationOfState{Float64}:
    ├── seawater_polynomial: TEOS10SeawaterPolynomial{Float64}
    └── reference_density: 1020.0
# Buoyancy that depends on both temperature and salinity
julia> buoyancy = SeawaterBuoyancy(equation_of_state=teos10)
SeawaterBuoyancy{Float64}:
├── gravitational_acceleration: 9.80665
└── equation of state: BoussinesqEquationOfState{Float64}
# Buoyancy that depends only on salinity with temperature held at 20 degrees Celsius
julia> salinity_dependent_buoyancy = SeawaterBuoyancy(equation_of_state=teos10, constant_temperature=20)
SeawaterBuoyancy{Float64}:
├── gravitational_acceleration: 9.80665
├── constant_temperature: 20
└── equation of state: BoussinesqEquationOfState{Float64}
# Buoyancy that depends only on temperature with salinity held at 35 psu
julia> temperature_dependent_buoyancy = SeawaterBuoyancy(equation_of_state=teos10, constant_salinity=35)
SeawaterBuoyancy{Float64}:
├── gravitational_acceleration: 9.80665
├── constant_salinity: 35
└── equation of state: BoussinesqEquationOfState{Float64}
```

File: src/BuoyancyModels/seawater_buoyancy.jl [hunk 3]
```diff
@@ -62,23 +62,39 @@ Examples
...
+julia> temperature_dependent_buoyancy = SeawaterBuoyancy(equation_of_state=teos10, constant_salinity=35)
+SeawaterBuoyancy{Float64}:
+├── gravitational_acceleration: 9.80665
+├── constant_salinity: 35
+└── equation of state: BoussinesqEquationOfState{Float64}
 
 # The "TEOS10" equation of state, see https://www.teos-10.org
+julia> using SeawaterPolynomials.TEOS10: TEOS10EquationOfState
+
 julia> teos10 = TEOS10EquationOfState()
```
Comment (glwagner) [inline src/BuoyancyModels/seawater_buoyancy.jl:98]: we don't need this again do we?

**Comment by navidcy** [comment]
Sorry, I had accidentally messed up and had the examples pasted twice!

**Comment by navidcy** [comment]
merge?

**Pull request merged**